### PR TITLE
Faq

### LINKED ISSUE #114: Create FAQ page
FAQ page with editable cards. These will be displayed to mobile?

### PR DESCRIPTION
**Pull request by AlexRutledge1**
# Describe your changes
Creates two new pages:
`/faq/edit`: only available if you are authenticated, allows the user to see the current FAQs and update/add/remove cards as they see fit
image
Users have to press `Deploy changes` to send the new list of FAQs to Firebase, they also have the option to discard their changes after they start editing.

The second page is available to everyone at `/faq` where you can freely see the FAQs that are stored in firebase. 
image

# Issue ticket number and link
closes #114

### CODE REVIEW

**Review by Henry-Graves — CHANGES_REQUESTED** [review]
when I click update, make no changes at all but save it, then discard my changes, I get stuck on "loading..." until I refresh.
The bug isn't there if you do actually make changes, so one fix is to not allow saving until the new string != old string.

As for UI changes, I think that "edit" may be a better word for users than "update", update could be read as something akin to refresh.
Also I think more visual distinction between the FAQ section and the discard/deploy buttons would benefit us. We could have more vertical padding between them, give the FAQ section a background, or something along those lines. It should make it more intuitive on what we're discarding/deploying.

this part: 
Screenshot (150)

File: src/pages/FAQ.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    const list = new Array<FAQItem>();
+    tempList.forEach( (val) => list.push(val));
+    setChangesExist(true);
+    setFAQCards(list);
+  };
+
+  const updateCard = (question: string, answer: string, index: number) => {
+    if (FAQCards === undefined)
+      return;
+    if (FAQCards.length <= index)
```
Comment (DanielW48) [inline src/pages/FAQ.tsx]: nit: index >= FAQCards.length for consistency pls

File: src/components/FAQCardBox.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  return (
+    <Box rounded='md' backgroundColor='primary.400' p='4'>
+      <HStack>Q:
+        <Input defaultValue={value.question} multiline onChangeText={setQuestion} numberOfLines={4} fontSize='lg'/>
+      </HStack>
+      <HStack>A:
+        <Input defaultValue={value.answer} multiline numberOfLines={4} onChangeText={setAnswer} fontSize='md'/>
+      </HStack>
+      <Button onPress={() => onChange(question, answer, index)} alignSelf='center'>
```
Comment (DanielW48) [inline src/components/FAQCardBox.tsx]: we should probably have a cancel button as well

File: src/pages/FAQ.tsx [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    setChangesExist(true);
+    setFAQCards(tempList);
+  };
+
+  const removeCard = (index: number) => {
+    if (FAQCards === undefined)
+      return;
+    if (index >= FAQCards.length)
+      return;
+    const tempList = FAQCards;
```
Comment (DanielW48) [inline src/pages/FAQ.tsx]: is this a reference copy? if so, we prolly need to do the splice on the new list we create, and then set it

File: src/pages/FAQ.tsx [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        data.map( (value, index) => 
+          <FAQCardBox key={index} question={value.question} answer={value.answer}/>)
+      }
+
+    </VStack>
+  );
+};
+
+export const EditFAQ = () => {
+  const [FAQCards, setFAQCards] = useState<FAQCollection>();
```
Comment (DanielW48) [inline src/pages/FAQ.tsx]: should this default to an empty list of FAQItem?

**Review by DanielW48 — COMMENTED** [review]
The branch has the old version of xplat pushed, I think you have some local submodule changes that need pushing

**Comment by AlexRutledge1** [comment]
> The branch has the old version of xplat pushed, I think you have some local submodule changes that need pushing

Yeah I was waiting for xplat to merge then add the new version.

File: src/pages/FAQ.tsx (hunk 4 above)
Comment (AlexRutledge1) [reply src/pages/FAQ.tsx]: Very true, good catch

**Comment by AlexRutledge1** [comment]
> 

New push should fix all this, check it out to be sure

**Review by Henry-Graves — APPROVED** [review]

**Pull request merged**